### PR TITLE
Refactor Item Exporters using Factory Method pattern for improved flexibility and maintainability

### PR DESCRIPTION
**Pull request by ANGELCHAVESCHINCHILLA**
# Description

This Pull Request introduces several significant changes to Scrapy's Item Exporters. Firstly, the Factory Method pattern has been implemented for creating item exporters. This provides a more flexible and maintainable structure for instantiating exporters, allowing for better scalability and extensibility in the future.

In addition to the Factory Method implementation, comprehensive unit tests have been added to ensure the proper functioning of the exporters. These tests ensure that the exporters are created correctly and produce the expected results in different scenarios.

Furthermore, the documentation of the Item Exporters has been modified to reflect the use of the Factory Method in instantiating the exporters. The documentation now provides clear instructions on how to instantiate and use the exporters using this pattern, making it easier for developers to understand and utilize them

# Benefits

The benefits of these changes are significant. The use of the Factory Method improves the modularity and flexibility of the exporters' code, enabling easy incorporation of new types of exporters in the future. Additionally, the addition of thorough unit tests ensures the quality and reliability of the exporters, facilitating maintenance and early detection of potential issues.

# Challenges

However, it is worth noting that these changes also presented certain challenges. The implementation of the Factory Method required careful refactoring of the existing code to ensure compatibility and consistency throughout the system. Moreover, creating comprehensive unit tests involved additional effort to ensure adequate coverage and the detection of possible errors.

# Changed location

Implementation of Factory Mehtod: [scrapy/exporters.py](https://github.com/scrapy/scrapy/blob/master/scrapy/exporters.py)

Unit testing: [tests/test_exporters.py](https://github.com/scrapy/scrapy/blob/master/tests/test_exporters.py)

Documentation:  [docs/topics/exporters.rst](https://github.com/scrapy/scrapy/blob/master/docs/topics/exporters.rst)

### CODE REVIEW

File: docs/topics/exporters.rst [hunk 1]
```diff
@@ -42,35 +42,36 @@ value of one of their fields:
 
 .. code-block:: python
 
-    from itemadapter import ItemAdapter
```
Comment (wRAR) [inline docs/topics/exporters.rst:45]: All of these whitespace changes look wrong.

File: docs/topics/exporters.rst [hunk 2]
```diff
@@ -42,35 +42,36 @@ value of one of their fields:
...
-    from scrapy.exporters import XmlItemExporter
+   from itemadapter import ItemAdapter
+   from item_exporters import ItemExporterFactory
```
Comment (wRAR) [inline docs/topics/exporters.rst:46]: There is no `item_exporters`.

File: docs/topics/exporters.rst (hunk 1 above)
Comment (wRAR) [reply docs/topics/exporters.rst:45]: As well as some unnecessary non-whitespace changes.

**Comment by wRAR** [comment]
Considering this is a second PR in two days that introduces unnecessary pattern implementations, is this some kind of school assignment?

File: scrapy/exporters.py [hunk 3]
```diff
@@ -363,3 +363,34 @@ def export_item(self, item):
         if self.binary:
             result = dict(self._serialize_item(result))
         return result
+
+
+class ItemExporterFactory:
```
Comment (wRAR) [inline scrapy/exporters.py:368]: I don't see how is this factory useful to be honest.

**Comment by ANGELCHAVESCHINCHILLA** [comment]
> Considering this is a second PR in two days that introduces unnecessary pattern implementations, is this some kind of school assignment?

File: scrapy/exporters.py (hunk 3 above)
Comment (ANGELCHAVESCHINCHILLA) [reply scrapy/exporters.py:368]: This PR has academic purposes (university project) but also allows the user to create a different exporter at runtime thanks to the type parameter

Comment (wRAR) [reply scrapy/exporters.py:368]: Please don't submit PRs that only have a purpose of demonstrating to your teacher that you wrote some code.

**Pull request closed without merging**